src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
+    def path: Path
+    def allBytes: Array[Byte]
+    def nodeType: fs.NodeType
+    def size: Long
+    def permissions: PosixFilePermissions
+    def atime: Long
+    def mtime: Long
+
+    def name : String    = path.getFileName.to_s
+    def blockCount: Long = (size + blockSize - 1) / blockSize
+    def blockSize: Long  = 512 // FIXME
+  }
+
+  // this could have been an apply method in an object allowing the use of Metadata instead of Metadataish
+  // I however could not get myself to repeat all those arguments
```

Comment:
I understand and agree with you. I was under the impression that we tried to get a few file systems coded up and take shortcuts. Once we have a few file systems ready we could create the correct abstractions.

If this is unacceptable to you, then I suggest you remodel the code in this branch so it reflects what you are thinking. It would take a lot of time and effort for me to get the abstractions as you want them. And please note that I want them the way you want them: you have put about 10 billion cycles of thought more into this than I have.

From the code that is here now (only a few examples of file systems) I can not yet device a good UX (in terms of traits, signatures, etc.). I am used to only writing code that is used and just keep expanding it, remodeling as a go. I wouldn't ship the code as it is now, but I feel we need more examples to continue.

So, let me know what you think and if it's easier to show in code, do that :-)